src/components/Details.jsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { setLoading, setPage, setRecipe } from '../redux/details/detailsSlice';
+
+const Details = () => {
+  const dispatch = useDispatch();
+  const { id } = useParams();
+  const { recipes } = useSelector((store) => store.recipes);
+  const { recipe, isLoading } = useSelector((store) => store.details);
+
+  useEffect(() => {
+    dispatch(setRecipe(recipes.find((recipe) => recipe.key === id)));
+    dispatch(setPage(recipe.label));
+    dispatch(setLoading(false));
+  }, [dispatch, id, recipes, recipe.label]);
+
+  if (isLoading) {
+    return (
+      <div className="loading">
+        <p>Loading details...</p>
+      </div>
+    );
+  }
+
+  return (
+    <div className="details">
+      <div className="headline">
+        <img className="recipe-img" src={recipe.images.SMALL.url} alt={recipe.label} />
+        <div className="text-group">
+          <p>{recipe.label}</p>
+          <p>{recipe.ingredients.length}</p>
+        </div>
+      </div>
+      <div className="divider">FULL RECIPE BELOW</div>
+      <div className="recipe-cont">
+        <div>
+          <div className="recipe-div recipe-url">
+            <h3>
+              See full recipe
+              <a href={recipe.url} target="_blank" rel="noreferrer noopener"> HERE.</a>
+            </h3>
+          </div>
+          <div className="recipe-div recipe-ingredients">
+            <h3>Ingredients</h3>
+            {recipe.ingredients.map((ingred) => <p key={ingred.foodId}>{ingred.text}</p>)}
+          </div>
+        </div>
+        <div className="recipe-nutrition">
+          <h3>Nutrition</h3>
```

Comment:
- [ ] As required by the requirements follow the same spacing, layouts, and design so you should display the details of the recipe in cards like the below design provided by Microverse in order to match it 
- screenshot of the requirements
<img width="544" alt="Screenshot 2023-04-19 at 22 03 36" src="https://user-images.githubusercontent.com/61790587/233209985-3212ef48-1f35-4881-8793-12b5ed137017.png">

- Yours 
<img width="306" alt="Screenshot 2023-04-19 at 22 21 41" src="https://user-images.githubusercontent.com/61790587/233213012-0b427931-e1cc-4483-9151-e6373e10dada.png">

- Design 
<img width="344" alt="Screenshot 2023-04-19 at 22 26 24" src="https://user-images.githubusercontent.com/61790587/233213186-29951a64-e359-4b70-aa1e-190621d683b9.png">

Reply:
Well resolved ✔️